src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -268,13 +285,15 @@ private List<HybridSearchCollector> getHybridSearchCollectors(final Collection<C
                         hybridSearchCollectors.add((HybridSearchCollector) sub);
                     }
                 }
-            } else if (collector instanceof HybridTopScoreDocCollector || collector instanceof HybridTopFieldDocSortCollector) {
-                hybridSearchCollectors.add((HybridSearchCollector) collector);
-            } else if (collector instanceof FilteredCollector
-                && (((FilteredCollector) collector).getCollector() instanceof HybridTopScoreDocCollector
-                    || ((FilteredCollector) collector).getCollector() instanceof HybridTopFieldDocSortCollector)) {
-                        hybridSearchCollectors.add((HybridSearchCollector) ((FilteredCollector) collector).getCollector());
-                    }
+            } else if (collector instanceof HybridTopScoreDocCollector
+                || collector instanceof HybridTopFieldDocSortCollector
+                || collector instanceof HybridCollapsingTopDocsCollector) {
```

Comment:
please extract condition inside this if into a separate method, this syntax became hard to read